consensus/polybft/checkpoint_manager.go

```diff
@@ -126,6 +126,16 @@ func (c *checkpointManager) submitCheckpoint(latestHeader *types.Header, isEndOf
 		return err
 	}
 
+	if lastCheckpointBlockNumber > latestHeader.Number {
+		// this node is syncing after its data was cleaned up
+		// what happened is that, node was the proposer on this block,
+		// and it already sent this checkpoint, but since its data got cleaned
+		// it started syncing from scratch, even its own minted blocks
+		// so there is no need to send a checkpoint, since it is already sent
+		// CheckpointManager will fail the transaction, so no need to spend tokens on it
```

Comment:
Let's shorten the comment (mind the linter, you would probably need to split it into several lines):

```suggestion
		// node is out of sync (doesn't have reached a tip of the chain), so even though it is a proposer, it would checkpoint block that is already checkpointed and transaction would fail anyway
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1982/commits/c4720038620cb297c87aca3d93003e334f237070